### PR TITLE
Rabbitmq

### PR DESCRIPTION
**Pull request by chayaZalaznik**

### CODE REVIEW

File: rabbitMQ/send_message.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    const message = {
+      data,
+      extraParam,
+    };
+    await channel.assertQueue(queue, { durable: false });
+    channel.sendToQueue(queue, Buffer.from(JSON.stringify(message)));
+    console.log(' [x] Sent \'%s\'', message);
+    await channel.close();
+  } catch (err) {
+    console.warn(err);
```
Comment (chanakamin) [inline rabbitMQ/send_message.js:18]: remove console

File: rabbitMQ/send_message.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  try {
+    connection = await amqp.connect('amqp://localhost');
+    const channel = await connection.createChannel();
+    const message = {
+      data,
+      extraParam,
+    };
+    await channel.assertQueue(queue, { durable: false });
+    channel.sendToQueue(queue, Buffer.from(JSON.stringify(message)));
+    console.log(' [x] Sent \'%s\'', message);
```
Comment (chanakamin) [inline rabbitMQ/send_message.js:15]: console...

File: rabbitMQ/ReceivingMessage.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    break;
+    // case 'addBackup':
+    //   addBackup(data);
+    //   break;
+  default:
+    logger.info(`Function ${functionName} not found.`);
+  }
+};
+
+export const startConsumer = async () => {
```
Comment (chanakamin) [inline rabbitMQ/ReceivingMessage.js:24]: where this method is called?

File: rabbitMQ/ReceivingMessage.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  const connection = await amqp.connect('amqp://localhost');
+  const channel = await connection.createChannel();
+  const queueName = 'website-subscriptions';
```
Comment (chanakamin) [inline rabbitMQ/ReceivingMessage.js:27]: save this value in a config file, and use it from there, here and in send_message file.

File: rabbitMQ/ReceivingMessage.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import amqp from 'amqplib';
+import addWebsite from '../controllers/website.js';
+import backupSite from '../Server.js';
+import logger from '../logger.js';
+
+export const handleData = async (data) => {
+  const functionName = data.extraParam;
```
Comment (chanakamin) [inline rabbitMQ/ReceivingMessage.js:7]: please, don't move the "functionName"
instead, you can send an "action"
its more clear. the name of the action and the function shouldn't be the same.

**Review by chanakamin — CHANGES_REQUESTED** [review]

File: app.js [hunk 6]
```diff
@@ -19,7 +19,7 @@ const app = express();
 
 dotenv.config();
 
-const port = process.env.PORT;
+const port = 3000;
```
[the remark this answers is not in the document]
Comment (chayaZalaznik) [reply app.js:23]: לא קשור עלינו זה שינוי קודם 
סתם שיניתי לו מיקום....

**Review by cheina1588 — CHANGES_REQUESTED** [review]

**Review by tamargit1 — APPROVED** [review]

**Review by tamargit1 — APPROVED** [review]

**Review by tamargit1 — APPROVED** [review]

**Review by tamargit1 — APPROVED** [review]

**Review by tamargit1 — APPROVED** [review]

**Review by cheina1588 — APPROVED** [review]

**Pull request merged**